test/test_datasets_download.py

```diff
@@ -43,89 +44,94 @@ def inner_wrapper(request, *args, **kwargs):
+            time.sleep(wait)
+    else:
+        raise AssertionError(
+            "\n".join(
+                (
+                    f"Assertion failed {times + 1} times with {wait:.1f} seconds intermediate wait time.\n",
+                    *(f"{idx}: {error}" for idx, error in enumerate(msgs, 1)),
                 )
             )
-
-    @staticmethod
-    def assert_response_ok(response, url=None, ok=200):
-        msg = f"The server returned status code {response.code}"
-        if url is not None:
-            msg += f"for the the URL {url}"
-        assert response.code == ok, msg
-
-    @staticmethod
-    def assert_is_downloadable(url):
-        request = Request(url, headers=dict(method="HEAD"))
-        response = urlopen(request)
-        DownloadTester.assert_response_ok(response, url)
-
-    @staticmethod
-    def assert_downloads_correctly(url, md5):
-        with get_tmp_dir() as root:
-            file = path.join(root, path.basename(url))
-            with urlopen(url) as response, open(file, "wb") as fh:
-                DownloadTester.assert_response_ok(response, url)
-                fh.write(response.read())
-
-            assert check_integrity(file, md5=md5), "The MD5 checksums mismatch"
-
-    def test_download(self):
-        assert_fn = (
-            lambda url, _: self.assert_is_downloadable(url)
-            if self.only_test_downloadability
-            else self.assert_downloads_correctly
         )
-        for url, md5 in self.collect_urls_and_md5s():
-            with self.subTest(url=url, md5=md5):
-                self.retry(lambda: assert_fn(url, md5))
 
-    def collect_urls_and_md5s(self):
-        raise NotImplementedError
 
-    @property
-    def only_test_downloadability(self):
-        return True
+def assert_server_response_ok(response, url=None):
+    msg = f"The server returned status code {response.code}"
+    if url is not None:
+        msg += f"for the the URL {url}"
+    assert 200 <= response.code < 300, msg
+
+
+def assert_url_is_accessible(url):
+    request = Request(url, headers=dict(method="HEAD"))
+    response = urlopen(request)
+    assert_server_response_ok(response, url)
+
+
+def assert_file_downloads_correctly(url, md5):
+    with get_tmp_dir() as root:
+        file = path.join(root, path.basename(url))
+        with urlopen(url) as response, open(file, "wb") as fh:
+            assert_server_response_ok(response, url)
+            fh.write(response.read())
+
+        assert check_integrity(file, md5=md5), "The MD5 checksums mismatch"
+
+
+class DownloadConfig:
+    def __init__(self, url, md5=None, id=None):
+        self.url = url
+        self.md5 = md5
+        self.id = id or url
+
+
+def make_parametrize_kwargs(download_configs):
+    argvalues = []
+    ids = []
+    for config in download_configs:
+        argvalues.append((config.url, config.md5))
+        ids.append(config.id)
+
+    return dict(argnames="url, md5", argvalues=argvalues, ids=ids)
+
+
+def places365():
```

Comment:
To add new download tests for a dataset, simply implement a similar function and add it to L130 and optionally to L135.